### PR TITLE
[Data Exfiltration Detection] Reduce False positives and add Dashboard to the package

### PR DESCRIPTION
**Pull request by sodhikirti07**
## What does this PR do?
This PR makes the following changes to this package:
- Updated datafeed of anomaly detection jobs to reduce false positives. 
  - Filtered out non-routable IPs
  - Filtered out "internal" and "inbound" connections
  - I haven't done anything to handle the duplicate records. We would need to add some aggregation or transform to normalize the duplicates.
- Updated `README.md` adding instructions to install dashboard
- Added `ML` tag and changed version in security rules
- Added dashboard 
- Made changes in `manifest.yml` to match latest formatting and updated `changelog.yml` 

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- https://github.com/elastic/security-ml/issues/211
- https://github.com/elastic/security-ml/issues/192

## Screenshots
For screenshots, refer to the comment below

### CODE REVIEW

**Comment by sodhikirti07** [comment]
ML jobs starting without errors:

image

Security rules enabled:

image

Dashboard installed:

image

Dashboard with data:

image

File: packages/ded/kibana/ml_module/ded-ml.json [hunk 1]
```diff
@@ -343,26 +360,43 @@
...
+            "bool": {
+                "must_not": [
+                          {
+                            "terms": {
+                              "network.direction": ["internal","inbound"]
+                            }
+                          },
+                          {
+                            "terms": {
+                              "source.ip": ["127.0.0.1"]
```
Comment (susan-shu-c) [inline packages/ded/kibana/ml_module/ded-ml.json:372]: So all of `127.0.0.1` will all be filtered out right? To check my understanding

Comment (sodhikirti07) [reply packages/ded/kibana/ml_module/ded-ml.json:372]: Yes, all will be filtered out. Also, I kept it an array in case we have other common IPs to add here.

**Comment by sodhikirti07** [comment]
Currently, we have three influencers in the anomaly detection jobs i.e. `host.name`, `user.name` and `process.name`. To make them more flexible, we decided to add `source.ip` and `destination.ip` based on this [comment](https://github.com/elastic/security-ml/issues/211#issuecomment-1407192103). With that said, I need your opinion on selecting the final set as three is the limit suggested by anomaly detection jobs (refer to this [doc](https://www.elastic.co/guide/en/machine-learning/current/ml-ad-run-jobs.html#ml-ad-influencers)).

**Comment by susan-shu-c** [comment]
That's a tough question!
At a first glance I'm thinking `host.name` and `process.name` and `destination.ip` might work, but all of them feel useful 😂 

The reason being I'm guessing that a malicious rare user might still be using a normal host name, but more likely to use an anomalous process? And then if they are doing data exfiltration they would likely have an anomalous destination ip, still.

Eagerly welcoming discussion on this from other folks too!

File: packages/ded/kibana/ml_module/ded-ml.json [hunk 2]
```diff
@@ -312,26 +312,43 @@
...
-                                    "exists": {
-                                        "field": "destination"
-                                    }
-                                }
-                            ]
+                            "bool": {
+                                "must_not": [
+                          {
+                            "terms": {
+                              "network.direction": ["internal","inbound"]
```
Comment (susan-shu-c) [inline packages/ded/kibana/ml_module/ded-ml.json:319]: Thanks for walking through this! I think this makes sense to me, given that we dug into the test environment's documents to double check for edge cases. And then we can ask Jake whom on Security we can ask more to double check the logic.

**Comment by ajosh0504** [comment]
@sodhikirti07 Thanks for doing this! It's looking great. Few qs:
- What's the duplicates issue you're referring to in the PR Description?
- Also, de we want to add source destination IPs as influencers to the anomaly detection jobs?

**Comment by sodhikirti07** [comment]
@ajosh0504 
> What's the duplicates issue you're referring to in the PR Description?

Duplicates are the same connection logs coming from multiple integrations. See this [comment](https://github.com/elastic/security-ml/issues/211#issuecomment-1409410630).

> Also, de we want to add source destination IPs as influencers to the anomaly detection jobs?
We should definitely add them as influencers. Just needed an opinion on the final set. So, shall we go ahead with `source.ip`, `destination.ip` and `process.name` as influencers.

**Comment by sodhikirti07** [comment]
@susan-shu-c I discussed this with Apoorva yesterday, and she said they had added influencers up to five in the past. So, we are okay on that front.

**Review by ajosh0504 — APPROVED** [review]
Looking great! 🚢

**Review by susan-shu-c — APPROVED** [review]
Thanks for walking through the logic!

File: packages/ded/docs/README.md [hunk 3]
```diff
@@ -42,5 +42,18 @@ To maximize the benefit of the Data Exfiltration Detection detection framework,
 | Potential Data Exfiltration Activity to an Unusual IP Address | An anomaly detection job has detected an abnormal volume of bytes being sent to an unusual IP address.|
 | Potential Data Exfiltration Activity to an Unusual Destination Port | An anomaly detection job has detected an abnormal volume of bytes being sent to an unusual destination port.|
 
+## Dashboard
+
+The **Data Exfiltration Detection Dashboard** is available under **Analytics > Dashboard**. This dashboard gives an overview of anomalies triggered for the data exfiltration detection package.
```
Comment (peteharverson) [inline packages/ded/docs/README.md:47]: Just wondering - this dashboard is not part of this package. How does it get installed? The recent lateral movement detection package has the [dashboard](https://github.com/elastic/integrations/blob/main/packages/lmd/kibana/dashboard/lmd-17fea180-8c4c-11ed-bb03-41a73f349362.json) as part of the package.

**Review by peteharverson — COMMENTED** [review]
The changes to the ML jobs LGTM, but I added a question about the dashboard.

File: packages/ded/docs/README.md (hunk 3 above)
Comment (sodhikirti07) [reply packages/ded/docs/README.md:47]: Thank you for pointing this out! Somehow, I forgot to push the update. I've committed the changes now, the dashboard file is [here](https://github.com/elastic/integrations/blob/d2742e6ab1d8216ce9673e9e7962e1363c92073c/packages/ded/kibana/dashboard/ded-c0d7f060-7a6b-11ed-9b26-4bb95b2c57a6.json).

Comment (susan-shu-c) [reply packages/ded/docs/README.md:47]: Thanks @peteharverson for noticing! The importance of a fresh pair of eyes; the applied ML team has been staring at these files for too long 😂

**Review by peteharverson — APPROVED** [review]
Thanks for adding the dashboard! LGTM.

Looking at the screenshot showing the ML job wizard, you could make all the job descriptions more concise by removing `A machine learning job to` leaving e.g. `Detects data exfiltration to an unusual geo-location (by city name).` . The first part of the description is redundant, but I know the jobs in the Lateral Movement Detection also follow this pattern. Just a thought :)
